diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6536e9a57..900376a45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,9 @@ cmake_minimum_required(VERSION 3.26)
 # If you want to only build one target, make sure to install it manually:
 # cmake --build . --target _C
 # cmake --install . --component _C
-project(vllm_extensions LANGUAGES CXX)
+project(vllm_extensions LANGUAGES CXX CUDA)
+enable_language(CUDA)
+find_package(CUDA REQUIRED)
 
 # CUDA by default, can be overridden by using -DVLLM_TARGET_DEVICE=... (used by setup.py)
 set(VLLM_TARGET_DEVICE "cuda" CACHE STRING "Target device backend for vLLM")
